### PR TITLE
Add Air Pollution

### PR DESCRIPTION
**Pull request by iomartin**
I decided to implement the air pollution data that I asked on #17 as it was pretty straightforward to add.

I can start working on adding to HA once this gets merged

### CODE REVIEW

File: src/pyopenweathermap/client/free_client.py [hunk 1]
```diff
@@ -14,6 +16,10 @@ def __init__(self, api_key, api_type, units="metric", lang='en'):
         self.api_type = api_type
         self.units = units
         self.lang = lang
+        if api_type == 'current':
```
Comment (freekode) [inline src/pyopenweathermap/client/free_client.py:19]: let's don't mix different clients together. air pollution client should be created in `owm_client_factory.py`

File: src/pyopenweathermap/client/onecall_client.py [hunk 2]
```diff
@@ -14,6 +16,7 @@ def __init__(self, api_key, api_version, units="metric", lang='en'):
         self.api_version = api_version
         self.units = units
         self.lang = lang
+        self.air_pollution = OWMAirPollutionClient(api_key, self.http_client)
```
Comment (freekode) [inline src/pyopenweathermap/client/onecall_client.py:19]: the same, shouldn't be here

File: tests/test_all.py [hunk 3]
```diff
@@ -18,6 +18,14 @@ async def test_api_30():
     assert report.current.date_time is not None
     assert report.hourly_forecast[0].condition.id is not None
     assert report.daily_forecast[0].condition.id is not None
+
+    report = await client.get_air_pollution(LATITUDE, LONGITUDE)
```
Comment (freekode) [inline tests/test_all.py:22]: tests are identical, it's better to extract it to separated method and remove duplicates. also by removing `current` and `forecast` only single test will remain

**Review by freekode — CHANGES_REQUESTED** [review]

**Comment by iomartin** [comment]
Thanks for the review @freekode, it will be a couple of days until I have some time to address your comments

**Comment by iomartin** [comment]
@freekode I've updated this PR addressing your comments. I made air_pollution_client.py a subclass of OWMClient(). Let me know if you don't like raising NotImplementedError() if someone for example calls get_weather() for the air pollution api. I thought it would be better than returning None, which could give the impression that something went wrong but was valid usage nonetheless.

Alternatively, we could rename get_weather() into something more generic like get_data() and then each subclass would only need to implement one function

**Comment by freekode** [comment]
LGTM Thanks
>if you don't like raising NotImplementedError()
Yeah, I think this is much better than return None.

>Alternatively, we could rename get_weather() into something more generic like get_data() and then each subclass would only need to implement one function
This is good idea. I even thought about to split it to get_current and get_forecast. So client can separate request for forecast.

**Pull request merged**